File: src/types/announcement.ts
import type { TimestampedEntity } from "./common";

export interface Announcement extends TimestampedEntity {
  title: string;
  message: string;
  eventDate?: string | null;
  eventTime?: string | null;
  venue?: string | null;
  imagePath?: string | null;
  category?: string | null;
  isActive: boolean;
}

export interface AnnouncementInput {
  title: string;
  message: string;
  eventDate?: string;
  eventTime?: string;
  venue?: string;
  imagePath?: string;
  category?: string;
  isActive?: boolean;
}

File: src-tauri/src/models/announcement.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Announcement {
    pub id: i64,
    pub title: String,
    pub message: String,
    pub event_date: Option<String>,
    pub event_time: Option<String>,
    pub venue: Option<String>,
    pub image_path: Option<String>,
    pub category: Option<String>,
    pub is_active: bool,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AnnouncementInput {
    pub title: String,
    pub message: String,
    pub event_date: Option<String>,
    pub event_time: Option<String>,
    pub venue: Option<String>,
    pub image_path: Option<String>,
    pub category: Option<String>,
    pub is_active: Option<bool>,
}
